Sources/HTTP/Responder/HTTPServer.swift

```diff
@@ -236,10 +236,26 @@ private final class HTTPServerHandler<R>: ChannelInboundHandler where R: HTTPSer
         // a valid request
         var reshead = res.head
         reshead.headers.add(name: "date", value: RFC1123DateCache.shared.currentTimestamp())
+
         if let server = serverHeader {
             reshead.headers.add(name: "server", value: server)
         }
 
+        // add 'Connection' header if needed
+        let connectionHeaders = reshead.headers[canonicalForm: "connection"].map { $0.lowercased() }
```

Comment:
you can make your life easier by just doing [`reqhead.isKeepAlive`](https://apple.github.io/swift-nio/docs/current/NIOHTTP1/Structs/HTTPRequestHead.html#/s:8NIOHTTP115HTTPRequestHeadV11isKeepAliveSbv) which will do a similar thing but also keep in mind if it's HTTP/1.0 (non-persistent which is like `connection: close` in HTTP/1.1) or HTTP/1.1 (defaults to persistent aka. keep-alive)